src/x_file.c

```diff
@@ -1308,7 +1358,38 @@ t_class *file_split_class,*file_join_class,*file_splitext_class, *file_splitname
         return do_file_handle_new(file_##verb##_class, s, argc, argv, verbose, creationmode); \
     }
 
-FILE_PD_NEW(handle, 1, 0666);
+static void file_define_ignore(t_file_handle*x, t_symbol*s, int argc, t_atom*argv) {
+        /* this is a noop (so the object does not bail out if somebody 'send's something to its label) */
+    (void)s;
+    (void)argc;
+    (void)argv;
+}
+static t_file_handle*file_define_new(t_symbol*s, int argc, t_atom*argv) {
+   t_file_handle*x = (t_file_handle*)pd_new(file_define_class);
+   x->x_fhptr = &x->x_fh;
+   x->x_fh.fh_fd = -1;
+   x->x_canvas = canvas_getcurrent();
+   x->x_creationmode = 0666;
+   x->x_verbose = 0;
+   if(1 == argc && A_SYMBOL == argv->a_type) {
+       x->x_fcname = atom_getsymbol(argv);
+       pd_bind(&x->x_obj.ob_pd, x->x_fcname);
+   } else {
+       pd_error(x, "%s requires an argument: handle name", s->s_name);
```

Comment:
This will print an error message whenever you open the help patch :-) `[text define]` and `[array define]` wouldn't complain and just set to an empty symbol.

Reply:
true.
but with `[text define]` resp `[array define]` you can actually *do something* with the object, even if there's no name attached to it.
whereas `[file define]` is useless without a name.

but yes, i probably could change this...

Reply:
or change the help-patch, to use `[file define $0.file]` instead...

Reply:
> but with [text define] resp [array define] you can actually do something with the object,

Yes, you can do *something*, but you can't do anything meaningful. All three objects expect a symbol to be actually useful, but only `[file define]` would complain if the symbol is missing. It's not too important, just an inconsistency.

Reply:
i've added an argument to the help-file so it doesn't through any warnings no more